src/request_handler.rs

```diff
@@ -1,7 +1,45 @@
 use hyper::{Body, Request, Response};
 
-pub fn request_entrypoint(req: Request<Body>) -> Response<Body> {
-    Response::new(Body::from(req.uri().path().to_string()))
+lazy_static! {
+    pub static ref GLOBAL_HANDLER: HttpRequestHandler = HttpRequestHandler::new();
+}
+
+pub fn global_request_entrypoint(req: Request<Body>) -> Response<Body> {
+    GLOBAL_HANDLER.handle(req)
+}
+
+#[derive(Debug)]
+pub struct HttpRequestHandler {}
+
+impl HttpRequestHandler {
+    fn new() -> HttpRequestHandler {
+        HttpRequestHandler {}
+    }
+
+    fn handle(&self, req: Request<Body>) -> Response<Body> {
+        let method = req.method();
+        let uri = req.uri();
+        let query = uri.query().unwrap_or("");
+
+        // Get the uri path, and then split it around slashes into components
+        let path_components: Vec<&str> = uri.path().split('/').skip(1).collect();
+
+        let response_body = if path_components[0] == "meta" {
```

Comment:
Might want to include a comment with a sample uri

Reply:
Will do!